kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/cache/Reflector.java

```diff
@@ -48,18 +47,18 @@
   private final AtomicBoolean isWatcherStarted;
   private final AtomicReference<Watch> watch;
 
-  public Reflector(Class<T> apiTypeClass, ListerWatcher<T, L> listerWatcher, Store store, OperationContext operationContext, long resyncPeriodMillis) {
+  public Reflector(Class<T> apiTypeClass, ListerWatcher<T, L> listerWatcher, Store store, OperationContext operationContext, long resyncPeriodMillis, ScheduledExecutorService resyncExecutor) {
     this.apiTypeClass = apiTypeClass;
     this.listerWatcher = listerWatcher;
     this.store = store;
     this.operationContext = operationContext;
     this.resyncPeriodMillis = resyncPeriodMillis;
-    lastSyncResourceVersion = new AtomicReference<>();
-    resyncExecutor = Executors.newSingleThreadScheduledExecutor();
-    watcher = new ReflectorWatcher<>(store, lastSyncResourceVersion, this::startWatcher, this::reListAndSync);
-    isActive = new AtomicBoolean(true);
-    isWatcherStarted = new AtomicBoolean(false);
-    watch = new AtomicReference<>(null);
+    this.lastSyncResourceVersion = new AtomicReference<>();
+    this.resyncExecutor = resyncExecutor;
```

Comment:
Is this ever not a `Executors.newSingleThreadScheduledExecutor()` ?

Reply:
Or maybe you forgot to pass the variable in Controller.java in lines 171 & 173?

Reply:
I exposed this so that I could mock this variable for testing. Do you want me to revert to old behavior?

Reply:
Ok, then maybe provide an extra constructor with the default value instead so we can simplify initialization in Controller method.